### DISCUSSION TITLE
Is it possible to get Chrome DevTools logs from performance testing?

### DISCUSSION THREAD

**Discussion by VicGrygorchyk** [webdriverio/webdriverio#6517] [General]
Hello
Is it possible to use Wd.io to get Chrome DevTools logs?
I am interested especially in 'Network' events [link](https://chromedevtools.github.io/devtools-protocol/tot/Network/)

Why Puppeteer 'intercept requests' is not enough for me? 
The interceptor only catches request before it was sent, while I want to get such events as `Network:requestWillBeSent `, what tells us, when request is ready to be sent, but might not be sent yet due to any blocking issue (so called `stalling time`) 

Why it should be possible to get such events with Wd.io?
According to https://webdriver.io/docs/devtools-service/#performance-testing Wd.io uses the lighthouse to get performance metrics. 
I am not sure aboud wdio implementation over the lighthouse, but if we use the lighthouse-core, we can get `Lhr` object after tests. This object contains `artifacts.devtoolsLogs`, what contains all Chrome DevTools Protocol logs and Network events as well.

If wd.io uses the ligthouse, is it posible to get the same raw `Lhr` object from the Wd.io for further parsing?
Maybe, there is another way to get such logs, please, advise me.
Regards,

**Comment by christian-bromann** [comment]
Hey @VicGrygorchyk 

WDIO offers a [`getPuppeteer`](https://webdriver.io/docs/api/browser/getPuppeteer) command that gives you access to the Puppeteer [browser](https://pptr.dev/#?product=Puppeteer&version=v7.1.0&show=api-class-browser) object. With that you have full access to CDP and its domains and listeners. See https://github.com/webdriverio/webdriverio/blob/main/examples/devtools/intercept.js as example.

Cheers

**Reply by VicGrygorchyk** [reply]
Did I get it right, the answer is "No" for the question :" is it posible to get the lighthouse Lhr object from the Wd.io" ?
Regards,

**Reply by christian-bromann** [reply]
What kind if Lhr object are you talking? It should be possible to expose every object we use.
